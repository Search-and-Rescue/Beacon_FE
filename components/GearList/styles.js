const styles = {
  gearListContainer: {
    alignItems: "center",
    height: "96%",
    marginTop: 20
  },
  gearList: {
    backgroundColor: "fff",
    borderColor: "#001028",
    borderRadius: 3,
    borderWidth: 4,
    flex: 1,
    height: 300,
    width: "90%"
  },
  gearCard: {
    backgroundColor: "#efefef",
    flex: 1,
    flexDirection: "row",
    height: 70,
    marginVertical: 5,
    justifyContent: "space-between"
  },
  itemRemoveBtn: {
    borderWidth: 1,
    borderColor: "#001028",
    height: 35,
    margin: 4,
    padding: 5
  },
  gearName: {
    flex: 1,
    fontSize: 21,
    paddingLeft: 5,
    paddingVertical: 8
  },
  addItemBtn: {
    borderColor: "black",
    borderWidth: 1,
    padding: 10,
    width: "100%"
  },
  addBtnText: {
    fontSize: 21,
    textAlign: "center"
  }
};

export default styles;
