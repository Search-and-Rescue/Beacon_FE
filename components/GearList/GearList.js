import React, { Component } from "react";
import { ScrollView, Text, View } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";
import { connect } from "react-redux";
import styles from "./styles";

export class GearList extends Component {
  constructor(props) {
    super(props);
  }

  editItem = gear => {
    this.props.navigation.navigate("Gear", { gear });
  };

  render() {
    const itemLinks = this.props.gear.map(gear => {
      return (
        <TouchableOpacity
          key={gear.id}
          style={styles.gearLink}
          onPress={() => this.editItem(gear)}
        >
          <Text style={styles.gearName}>{gear.itemName}</Text>
        </TouchableOpacity>
      );
    });

    return (
      <View>
        <View style={styles.gearListContainer}>
          <ScrollView style={styles.gearList}>{itemLinks}</ScrollView>

          <TouchableOpacity
            style={styles.addItemBtn}
            onPress={() => this.props.navigation.navigate("Gear", {gear:{}})}
          >
            <Text style={styles.addBtnText}>Add a Gear Item</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}

export const mapStateToProps = ({ gear }) => ({
  gear
});

export default connect(mapStateToProps)(GearList);
