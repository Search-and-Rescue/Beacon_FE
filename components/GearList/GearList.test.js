import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { GearList } from './GearList';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

jest.mock('react-native-gesture-handler', () => 'TouchableOpacity');

jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: ""
}));

describe('GearList', () => {
  let wrapper;

  beforeEach(() => {
    const mockGear = [{
      id: 89,
      itemName: 'Rustic Iron Bottle',
      description: 'Chuck Norris can binary search unsorted data.'
    }];

    wrapper = shallow(<GearList
      gear={mockGear}
    />);
  });

  it('should match the snapshot with all of the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });
});