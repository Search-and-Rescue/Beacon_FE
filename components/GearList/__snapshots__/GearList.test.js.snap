// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GearList should match the snapshot with all of the data passed through 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "96%",
        "marginTop": 20,
      }
    }
  >
    <ImageBackground
      imageStyle={
        Object {
          "opacity": 0.2,
        }
      }
      source={
        Object {
          "testUri": "../../../assets/background.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "width": "100%",
        }
      }
    >
      <ScrollViewMock
        style={
          Object {
            "backgroundColor": "fff",
            "borderColor": "#001028",
            "borderRadius": 3,
            "borderWidth": 4,
            "flex": 1,
            "height": 300,
            "marginHorizontal": "5%",
            "width": "90%",
          }
        }
      >
        <View
          key="89"
          style={
            Object {
              "backgroundColor": "#efefef",
              "flex": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "marginVertical": 5,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "#001028",
                "borderColor": "#001028",
                "borderWidth": 1,
                "color": "#efefef",
                "height": 30,
                "marginLeft": 7,
                "marginVertical": 10,
                "padding": 5,
              }
            }
          >
            REMOVE
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontFamily": "Futura",
                "fontSize": 21,
                "paddingLeft": 7,
                "paddingVertical": 10,
              }
            }
          >
            Rustic Iron Bottle
          </Text>
        </View>
      </ScrollViewMock>
    </ImageBackground>
  </View>
</View>
`;
